src/onCLS.ts

```diff
@@ -49,99 +49,88 @@ export const onCLS = (onReport: CLSReportCallback, opts?: ReportOpts) => {
+            // If the entry occurred less than 1 second after the previous entry
+            // and less than 5 seconds after the first entry in the session,
+            // include the entry in the current session. Otherwise, start a new
+            // session.
+            if (
+              sessionValue &&
+              entry.startTime - lastSessionEntry.startTime < 1000 &&
+              entry.startTime - firstSessionEntry.startTime < 5000
+            ) {
+              sessionValue += entry.value;
+              sessionEntries.push(entry);
+            } else {
+              sessionValue = entry.value;
+              sessionEntries = [entry];
+            }
           }
+        });
 
-          // If the current session value is larger than the current CLS value,
-          // update CLS and the entries contributing to it.
-          if (sessionValue > metric.value) {
-            metric.value = sessionValue;
-            metric.entries = sessionEntries;
-            report();
-          }
-        }
-      });
-    };
-
-    const po = observe('layout-shift', handleEntries);
-    if (po) {
-      report = bindReporter(
-        onReportWrapped,
-        metric,
-        thresholds,
-        opts!.reportAllChanges
-      );
-
-      // Start monitoring FCP so we can only report CLS if FCP is also reported.
-      // Note: this is done to match the current behavior of CrUX.
-      // Also, if there have not been any layout shifts when FCP is dispatched,
-      // call "report" with a zero value
-      onFCP((fcpMetric) => {
-        fcpValue = fcpMetric.value;
-        if (metric.value < 0) {
-          metric.value = 0;
+        // If the current session value is larger than the current CLS value,
+        // update CLS and the entries contributing to it.
+        if (sessionValue > metric.value) {
+          metric.value = sessionValue;
+          metric.entries = sessionEntries;
           report();
         }
-      });
-
-      onHidden(() => {
-        handleEntries(po.takeRecords() as CLSMetric['entries']);
-        report(true);
-      });
+      };
 
-      // Only report after a bfcache restore if the `PerformanceObserver`
-      // successfully registered.
-      onBFCacheRestore(() => {
-        sessionValue = 0;
-        fcpValue = -1;
-        metric = initMetric('CLS', 0);
+      const po = observe('layout-shift', handleEntries);
+      if (po) {
         report = bindReporter(
-          onReportWrapped,
+          onReport,
           metric,
           thresholds,
           opts!.reportAllChanges
         );
 
-        doubleRAF(() => report());
-      });
-    }
-  });
+        onHidden(() => {
+          handleEntries(po.takeRecords() as CLSMetric['entries']);
+          report(true);
+        });
+
+        // Only report after a bfcache restore if the `PerformanceObserver`
+        // successfully registered.
+        onBFCacheRestore(() => {
+          sessionValue = 0;
+          metric = initMetric('CLS', 0);
```

Comment:
I think it could, but I don't think there's any good way to prevent it. We have another similar problem today where someone could load a page and then switch tabs away and then back again before the web-vitals library loads, and we can't detect those cases properly right now in JS (though they are tracked properly in UKM).

The approach I've been taking is to be as accurate as possible for now, without going overboard  with extra complexity, and wait until we eventually have native APIs for all of these edge cases (e.g. [`visibility-state` performance entries](https://chromestatus.com/feature/5683502144028672)).